gaeguli/pipeline.c

```diff
@@ -984,6 +988,75 @@ gaeguli_pipeline_add_srt_target_full (GaeguliPipeline * self,
   return NULL;
 }
 
+GaeguliRecordTarget *
+gaeguli_pipeline_add_record_target_full (GaeguliPipeline * self,
+    GaeguliVideoCodec codec, guint bitrate, const gchar * location,
+    GError ** error)
+{
+  guint target_id = 0;
+  GaeguliRecordTarget *rec_target = NULL;
+  g_autoptr (GstPad) tee_srcpad = NULL;
+
+  g_return_val_if_fail (GAEGULI_IS_PIPELINE (self), 0);
+  g_return_val_if_fail (location != NULL, 0);
+  g_return_val_if_fail (error == NULL || *error == NULL, 0);
+
+  g_mutex_lock (&self->lock);
+
+  if (self->vsrc == NULL) {
+    goto failed;
+  }
+
+  target_id = g_str_hash (location);
+
+  rec_target =
+      g_hash_table_lookup (self->rec_targets, GINT_TO_POINTER (target_id));
+
+  if (!rec_target) {
+    g_autoptr (GstElement) tee = NULL;
+    g_autoptr (GstPad) tee_srcpad = NULL;
+    g_autoptr (GError) internal_err = NULL;
+
+    g_debug ("no target pipeline mapped with [%x]", target_id);
+
+    tee = gst_bin_get_by_name (GST_BIN (self->vsrc), "tee");
+    tee_srcpad = gst_element_request_pad (tee,
+        gst_element_class_get_pad_template (GST_ELEMENT_GET_CLASS (tee),
+            "src_%u"), NULL, NULL);
+
+    rec_target =
+        gaeguli_record_target_new (tee_srcpad, target_id, codec, bitrate,
+        self->fps, location, &internal_err);
+
+    if (rec_target == NULL) {
+      g_propagate_error (error, internal_err);
+      internal_err = NULL;
+      goto failed;
+    }
+
+    g_hash_table_insert (self->rec_targets, GINT_TO_POINTER (target_id),
+        rec_target);
+  }
```

Comment:
If a programmer uses the same location for different recording targets by mistake, it will be silently ignored.
debug or warning messages are required if the location is duplicated.